packages/common/src/lib/common/tools.ts

```diff
@@ -3143,6 +3153,22 @@ export function pipeLinewise(input: NodeJS.ReadableStream, output: NodeJS.Writab
     });
 }
 
+/**
+ * Checks if an adapter is an ESM module or CJS
+ *
+ * @param adapter name of the adapter like hm-rpc
+ */
+export async function isAdapterEsmModule(adapter: string): Promise<boolean> {
+    const adapterDir = getAdapterDir(adapter);
+    if (!adapterDir) {
+        throw new Error(`Could not find adapter dir of ${adapter}`);
+    }
+
+    const packJson = await fs.readJSON(path.join(adapterDir, 'package.json'), { encoding: 'utf-8' });
```

Comment:
in theory the adapter main file could lie in another dir having a separate package.json with a separate type argument, but I think nobody should do this..